v0/destinations/marketo/transform.js

```diff
@@ -328,16 +350,42 @@ const processIdentify = async (message, formattedDestination, token) => {
   if (isDefinedAndNotNull(userId)) {
     inputObj.userId = userId;
   }
+  let endPoint = `https://${accountId}.mktorest.com/rest/v1/leads.json`;
+  let payload = {
+    action: "createOrUpdate",
+    input: [inputObj],
+    lookupField: "id"
+  };
+  // handled if vdm enabled
+  if (get(message, MappedToDestinationKey)) {
+    const { objectType } = getDestinationExternalIDInfoForRetl(
+      message,
+      "MARKETO"
+    );
+    // if leads object then will fallback to the already existing endpoint and payload
+    if (objectType !== "leads") {
+      endPoint = `https://${accountId}.mktorest.com/rest/v1/customobjects/${objectType}.json`;
+      // we will be using the dedupeBy dedupeFields for this endpoint
+      // DOC: https://developers.marketo.com/rest-api/lead-database/custom-objects/#create_and_update
+      // if marketoGUID is mapped it should be removed before sending to marketo as for this type the following error can arise
+      // Field 'marketoGUID' not updateable or Field 'marketoGUID' is not allowed when dedupeBy is 'dedupeFields'
+      if (traits.marketoGUID) {
+        delete traits.marketoGUID;
+      }
+      const input = removeUndefinedValues(traits);
+      payload = {
+        action: "createOrUpdate",
+        dedupeBy: "dedupeFields",
```

Comment:
What if the dedupeField is not present in input are we handling that scenario?

Reply:
it is required from vdm to map them so it will be present in traits.

Reply:
Ok but as we had custom objects as a requirement from customer can we make it generic (limit it not only to vdm)
Example normal identify call?

Reply:
For making it generic we need to know which object the user wants to send this too. How are we gonna plan this right now? Through ui/externalId or something else?

Reply:
As this should be event specific I am thinking externalId support should be done. Ok for VDM requirements this is sufficient

Reply:
Yeah we can come up with a plan for this.